### PR TITLE
Use phpdbg to create coverage data

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
Use phpdbg to generate test coverage ... should be much faster compared to xdebug ...

## Motivation and Context
Developers shall see the change in test code coverage within the pull request.

## ToDos
- [ ] set proper branch - needs Jenkins magic
- [ ] could we already run this isolated on drone? Hopefully drone knows the proper branch name?

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by tboerger — APPROVED** [review]
Than you can also drop the separate coverage report jobs, right?

**Comment by DeepDiver1975** [comment]
> Than you can also drop the separate coverage report jobs, right?

yes - already disabled ...

but as discussed I'll move this to drone right away

File: .drone.yml [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    pull: true
+    secrets: [ cache_s3_endpoint, cache_s3_access_key, cache_s3_secret_key ]
+    rebuild: true
+    mount:
+      - lib/composer
+      - core/vendor
+      - build/node_modules
+    when:
+      local: false
+      branch: [ master, stable10, stable9.1, stable9, stable8.2 ]
```
Comment (tboerger) [inline .drone.yml]: Drop `stable8.2`

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+      event: [push, pull_request]
+      branch: [master, stable10, stable9.1, stable9]
+      matrix:
+        TEST_SUITE: coverage
+
+  codecov:
+    image: robertstettner/drone-codecov
+    secrets: [codecov_token]
+    pull: true
+    environment:
```
Comment (tboerger) [inline .drone.yml]: You can drop this `environment:` part

File: .drone.yml [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    commands:
+      - ./tests/drone/test-coverage.sh
+    when:
+      event: [push, pull_request]
+      branch: [master, stable10, stable9.1, stable9]
+      matrix:
+        TEST_SUITE: coverage
+
+  codecov:
+    image: robertstettner/drone-codecov
```
Comment (tboerger) [inline .drone.yml]: Replace this with `plugins/codecov:1`, I have just pushed that.

File: .drone.yml [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    secrets: [ cache_s3_endpoint, cache_s3_access_key, cache_s3_secret_key ]
+    rebuild: true
+    mount:
+      - lib/composer
+      - core/vendor
+      - build/node_modules
+    when:
+      local: false
+      branch: [ master, stable10, stable9.1, stable9 ]
+      event: [ push ]
```
Comment (tboerger) [inline .drone.yml]: Move one line up

File: .drone.yml [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+
+  flush:
+    image: plugins/s3-cache:1
+    pull: true
+    secrets: [ cache_s3_endpoint, cache_s3_access_key, cache_s3_secret_key ]
+    flush: true
+    flush_age: 14
+    when:
+      local: false
+      event: [push, pull_request]
```
Comment (tboerger) [inline .drone.yml]: Only on push event

**Review by tboerger — COMMENTED** [review]
These 2 nitpick comments than it's fine to merge

**Review by tboerger — APPROVED** [review]

**Pull request merged**